### PR TITLE
Add support for partial syncs in sync_local_to_changes

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: dash/utils/sync.py [hunk 1]
```diff
@@ -184,15 +185,10 @@ def sync_local_to_set(org, syncer, remote_set):
             syncer.delete_local(local)
             outcome_counts[SyncOutcome.deleted] += 1
 
-    return (
-        outcome_counts[SyncOutcome.created],
-        outcome_counts[SyncOutcome.updated],
-        outcome_counts[SyncOutcome.deleted],
-        outcome_counts[SyncOutcome.ignored],
-    )
+    return outcome_counts
```
Comment (rowanseymour) [inline dash/utils/sync.py:188]: changing this signature for consistency - it also returns the counts as a dict now

File: dash/utils/sync.py [hunk 2]
```diff
@@ -184,15 +185,10 @@ def sync_local_to_set(org, syncer, remote_set):
...
 
 
-def sync_local_to_changes(org, syncer, fetches, deleted_fetches, progress_callback=None):
+def sync_local_to_changes(org, syncer, fetches, deleted_fetches, progress_callback=None, time_limit : int = None) -> Tuple[dict, Optional[str]]:
```
Comment (rowanseymour) [inline dash/utils/sync.py:191]: didn't want to add a 5th return value so now counts stay packed in the dict

File: dash/utils/sync.py (hunk 1 above)
Comment (norkans7) [reply dash/utils/sync.py:188]: This require we import `SyncOutcome` to use that dict returned 
Isn't it better to have string keys here instead?

Comment (rowanseymour) [reply dash/utils/sync.py:188]: Usually I like constants but yeah this could be annoying since casepro wants to serialize these as JSON

Comment (rowanseymour) [reply dash/utils/sync.py:188]: Hmm tho other methods in this module like `sync_from_remote` already return `SyncOutcome` values

Comment (norkans7) [reply dash/utils/sync.py:188]: ok

**Pull request merged**